allennlp/training/metrics/bleu.py

```diff
@@ -134,7 +135,29 @@ def __call__(
             self._reference_lengths += valid_gold_targets_mask.sum().item()
 
     @overrides
-    def get_metric(self, reset: bool = False) -> Dict[str, float]:
+    def get_metric(
+        self,
+        reset: bool = False,
+        world_size: int = 1,
+        cuda_device: Union[int, torch.device] = torch.device("cpu"),
+    ) -> Dict[str, float]:
+
+        if world_size > 1:
+            _prediction_lengths = torch.tensor(self._prediction_lengths).to(cuda_device)
+            _reference_lengths = torch.tensor(self._reference_lengths).to(cuda_device)
+            for ngram in self._precision_matches:
+                _precision_matches = torch.tensor(self._precision_matches[ngram]).to(cuda_device)
+                _precision_totals = torch.tensor(self._precision_totals[ngram]).to(cuda_device)
+                dist.all_reduce(_precision_matches, op=dist.ReduceOp.SUM)
+                dist.all_reduce(_precision_totals, op=dist.ReduceOp.SUM)
+                self._precision_matches[ngram] = _precision_matches.item() / world_size
+                self._precision_totals[ngram] = _precision_totals.item() / world_size
```

Comment:
Is that correct? What if one worker saw an ngram 100 times, and another saw it only once?